### PR TITLE
inccommand: Ignore errors during preview and make cmd_can_preview stricter

### PR DESCRIPTION
**Pull request by jamessan**

### CODE REVIEW

File: src/nvim/ex_getln.c [hunk 1]
```diff
@@ -1602,12 +1602,16 @@ static int command_line_changed(CommandLineState *s)
...
+    //    0. Block error reporting since the command may not yet be functional
     //    1. Do the command.
     //    2. Command implementation detects CMDPREVIEW state, then:
     //       - Update the screen while the effects are in place.
     //       - Immediately undo the effects.
+    //    3. Unblock error reporting
     State |= CMDPREVIEW;
+    emsg_silent++;
     do_cmdline(ccline.cmdbuff, NULL, NULL, DOCMD_KEEPLINE|DOCMD_NOWAIT);
+    emsg_silent--;
```
Comment (justinmk) [inline src/nvim/ex_getln.c]: I think the `emsg_off` pair in `ex_substitute()` can removed now.

**Comment by justinmk** [comment]
Looks great. Also add `Closes #5912` to the commit?

File: src/nvim/ex_getln.c (hunk 1 above)
Comment (jamessan) [inline src/nvim/ex_getln.c]: Good call.  Updated.

File: src/nvim/ex_getln.c [hunk 2]
```diff
@@ -1602,12 +1602,16 @@ static int command_line_changed(CommandLineState *s)
              && cmd_can_preview(ccline.cmdbuff)
              && !vpeekc_any()) {
     // Show 'inccommand' preview. It works like this:
+    //    0. Block error reporting since the command may not yet be functional
```
Comment (justinmk) [inline src/nvim/ex_getln.c]: this and its counterpart seems like an implementation detail, maybe inline it next to the respective increment/decrement.

File: test/functional/ui/inccommand_spec.lua [hunk 3]
```diff
@@ -207,6 +208,42 @@ describe(":substitute, 'inccommand' preserves", function()
...
+  for _, case in pairs{"", "split", "nosplit"} do
+    it("visual selection for non-previewable command (inccommand="..case..") #5888", function()
+      local screen = Screen.new(30,10)
+      common_setup(screen, case, default_text)
+      feed('1G2V')
+
+      feed(':s')
+      screen:expect([[
+        {vis:Inc substitution on}           |
+        t{vis:wo lines}                     |
```
Comment (justinmk) [inline test/functional/ui/inccommand_spec.lua]: if you change it to `{vis:two lines}` does the test fail? That seems weird.

Comment (jamessan) [inline test/functional/ui/inccommand_spec.lua]: Because that's where the cursor is drawn.

**Review by justinmk — APPROVED** [review]

**Pull request merged**